### PR TITLE
update revapi of spring package for main branch

### PR DESCRIPTION
**Pull request by Netyyyy**
Remove suppression of spring package and add targeted suppression.

### CODE REVIEW

File: eng/code-quality-reports/src/main/resources/revapi/revapi.json [hunk 1]
```diff
@@ -109,6 +105,36 @@
...
+          "code": "java.method.added",
+          "new": "method (java.util.function.Supplier<java.lang.String>|void) com.azure.communication.common.CommunicationTokenRefreshOptions..*",
+          "justification": "method allowed to add."
+        },
+        {
+          "regex": true,
+          "code": "java.method.added",
+          "new": "method com.azure.communication.common..*",
+          "justification": "method allowed to add."
+        },
```
Comment (saragluna) [inline eng/code-quality-reports/src/main/resources/revapi/revapi.json:119]: Why are these two needed? I remember we don't use these libraries.

File: eng/code-quality-reports/src/main/resources/revapi/revapi.json [hunk 2]
```diff
@@ -109,6 +105,36 @@
...
+          "code": "java.class.nonPublicPartOfAPI",
+          "new": "(class|interface) org.springframework..*",
+          "justification": "We allow this in springframework"
+        },
+        {
+          "regex": true,
+          "code": "java.class.nonPublicPartOfAPI",
+          "new": "class com.azure.spring..*",
+          "justification": "We allow this in azure.spring"
+        },
```
Comment (saragluna) [inline eng/code-quality-reports/src/main/resources/revapi/revapi.json:137]: What is this rule used for?

**Review by saragluna — CHANGES_REQUESTED** [review]

**Review by alzimmermsft — APPROVED** [review]

**Pull request merged**